packages/builders/src/messages/embed/UnsafeEmbed.ts

```diff
@@ -1,101 +1,144 @@
-import type {
-	APIEmbed,
-	APIEmbedAuthor,
-	APIEmbedField,
-	APIEmbedFooter,
-	APIEmbedImage,
-	APIEmbedProvider,
-	APIEmbedThumbnail,
-	APIEmbedVideo,
-} from 'discord-api-types/v9';
+import type { APIEmbed, APIEmbedField, APIEmbedVideo } from 'discord-api-types/v9';
 import { Embed } from './Embed';
 
-export interface AuthorOptions {
+export interface EmbedAuthorData {
 	name: string;
 	url?: string;
 	iconURL?: string;
+	proxyIconURL?: string;
 }
 
-export interface FooterOptions {
+export type EmbedAuthorOptions = Omit<EmbedAuthorData, 'proxyIconURL'>;
+
+export interface EmbedFooterData {
 	text: string;
 	iconURL?: string;
+	proxyIconURL?: string;
+}
+
+export type EmbedFooterOptions = Omit<EmbedFooterData, 'proxyIconURL'>;
+
+export interface EmbedImageData {
+	url: string;
+	proxyURL?: string;
+	height?: number;
+	width?: number;
 }
 
-export class UnsafeEmbed implements APIEmbed {
+export class UnsafeEmbed {
+	protected data: APIEmbed;
+
+	public constructor(data: APIEmbed = {}) {
+		this.data = { fields: [], ...data };
+		if (data.timestamp) this.data.timestamp = new Date(data.timestamp).toISOString();
+	}
+
 	/**
 	 * An array of fields of this embed
 	 */
-	public readonly fields: APIEmbedField[];
+	public get fields() {
+		return this.data.fields ?? [];
```

Comment:
Is the `?? []` needed? We're always ensuring an array in the constructor.

Reply:
Alternatively, we could just not initialise fields (but keep `this.data = { ...data }` to clone the object) and then create the array if it doesn't exist in the field methods, this way we save some memory when initialising embeds since we won't be instantiating an array all the time, even for objects that don't have them.